sparse_list.py

```diff
@@ -69,14 +69,26 @@ def __setslice__(self, start, stop, vals):
     def __delitem__(self, item):
         if isinstance(item, slice):
             indices = xrange(*item.indices(self.size))
+        elif item < 0:
+            indices = (self.size + item, )
         else:
             indices = (item, )
 
-        for i in indices:
-            try:
-                del self.elements[i]
-            except KeyError:
-                pass
+        offset = 0
+
+        for k in list(self.elements.keys()):
```

Comment:
Why convert to a list?
I think you need to ensure that you are browsing the keys in increasing order, otherwise the computation of the `offset` is likely to go wrong. So I would replace `list` by `sorted` here.

Reply:
...because Python 3 will raise a `RuntimeError: dictionary changed size during iteration` without it.

Reply:
`sorted()` would be safer, though 👍